src/libraries/System.Net.Http/ref/System.Net.Http.cs

```diff
@@ -212,13 +214,28 @@ public partial class HttpRequestMessage : System.IDisposable
         public System.Net.Http.HttpContent? Content { get { throw null; } set { } }
         public System.Net.Http.Headers.HttpRequestHeaders Headers { get { throw null; } }
         public System.Net.Http.HttpMethod Method { get { throw null; } set { } }
+        [Obsolete("Use Options instead.")]
         public System.Collections.Generic.IDictionary<string, object?> Properties { get { throw null; } }
+        public HttpRequestOptions Options { get { throw null; } }
         public System.Uri? RequestUri { get { throw null; } set { } }
         public System.Version Version { get { throw null; } set { } }
         public void Dispose() { }
         protected virtual void Dispose(bool disposing) { }
         public override string ToString() { throw null; }
     }
+
+    public readonly struct HttpRequestOptionsKey<TValue>
+    {
+        public HttpRequestOptionsKey(string key) {}
+        public string Key { get { throw null; } }
+    }
+
+    public sealed class HttpRequestOptions : System.Collections.Generic.Dictionary<string, object> 
```

Comment:
According to the proposal `HttpRequestOptions` should implement `IDictionary<string, object?>` explicitly.

```c#
    public sealed class HttpRequestOptions : IDictionary<string, object?>
    {
        // Explicit interface implementation
        public bool TryGetValue<TValue>(HttpRequestOptionsKey<TValue> key, out TValue value);
        public void Set<TValue>(HttpRequestOptionsKey<TValue> key, TValue value);
    }
```

Reply:
Addressed in recent commit